components/grid/export/pdf.md

```diff
@@ -10,17 +10,223 @@ position: 1
+
+@using System.IO
 
-* <a href="https://github.com/telerik/blazor-ui/tree/master/grid/pdf-export-server" target="_blank">Export Grid to PDF on the Server</a> - gets the `DataSourceRequest` from the grid and sends it to the server service for processing. It is used to let you fetch the same data that the grid has (including paging, filtering, sorting) so you can generate the PDF. For WebAssembly apps, this improves the performance by not generating the file in the browser, which is, at the time of writing, too slow for a real-life scenario.
+<TelerikButton OnClick="@(async () => await GridRef.SaveAsPdfFileAsync())">Download the PDF file</TelerikButton>
+<TelerikButton OnClick="@GetTheDataAsAStream">Get the Exported Data as a MemoryStream</TelerikButton>
 
-* <a href="https://github.com/telerik/blazor-ui/tree/master/common/pdf-jpg-export-js" target="_blank">PDF and JPG Export in the Browser with JS</a> - uses Kendo JS libraries to generate the PDF file from the current DOM in the browser.
+<TelerikGrid @ref="@GridRef"
+             Data="@GridData"
+             FilterMode="@GridFilterMode.FilterMenu"
+             Pageable="true"
+             Sortable="true"
+             Reorderable="true"
+             Resizable="true"
+             Groupable="true"
+             Width="650px">
+
+    <GridToolBarTemplate>
+        <GridCommandButton Command="PdfExport" Icon="@SvgIcon.FilePdf">Export to PDF</GridCommandButton>
+        <label class="k-checkbox-label"><TelerikCheckBox @bind-Value="@ExportAllPages" />Export All Pages</label>
+    </GridToolBarTemplate>
+
+    <GridExport>
+        <GridPdfExport FileName="telerik-grid-export" AllPages="@ExportAllPages" />
+    </GridExport>
+
+    <GridColumns>
+        <GridColumn Field="@nameof(SampleData.ProductId)" Title="ID" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.ProductName)" Title="Product Name" Width="200px" />
+        <GridColumn Field="@nameof(SampleData.UnitsInStock)" Title="In stock" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.Price)" Title="Unit Price" Width="100px" />
+        <GridColumn Field="@nameof(SampleData.Discontinued)" Title="Discontinued" Width="150px" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private TelerikGrid<SampleData> GridRef { get; set; }
+
+    private MemoryStream exportedPdfStream { get; set; }
+
+    private List<SampleData> GridData { get; set; }
+
+    private bool ExportAllPages { get; set; }
+
+    private async Task GetTheDataAsAStream()
+    {
+        MemoryStream finalizedStream = await GridRef.ExportToPdfAsync();
+
+        exportedPdfStream = new MemoryStream(finalizedStream.ToArray());
+    }
+
+    protected override void OnInitialized()
+    {
+        GridData = Enumerable.Range(1, 100).Select(x => new SampleData
+            {
+                ProductId = x,
+                ProductName = $"Product {x}",
+                UnitsInStock = x * 2,
+                Price = 3.14159m * x,
```

Comment:
Why so much precision? You can also use `Random.Shared.Next()`

Reply:
Random.Shared.Next( ) returns an integer, while this is a more "realistic" price value. We can indeed reduce the precision a bit, though, will do.